satellite/satellitedb/containment.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type containment struct {
+	db *dbx.DB
+}
+
+// Get gets the pending audit by node id
+func (containment *containment) Get(ctx context.Context, id pb.NodeID) (*audit.PendingAudit, error) {
+	if id.IsZero() {
+		return nil, audit.ContainError.New("node ID empty")
+	}
+
+	pending, err := containment.db.Get_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return nil, audit.ErrContainedNotFound.New(id.String())
+	}
+	if err != nil {
+		return nil, audit.ContainError.Wrap(err)
+	}
+
+	return convertDBPending(pending)
+}
+
+// IncrementPending will create a new pending audit entry, or increase its reverify count if it already exists
+func (containment *containment) IncrementPending(ctx context.Context, pendingAudit *audit.PendingAudit) error {
+	statement := containment.db.Rebind(
+		`INSERT INTO pending_audits (node_id, piece_id, stripe_index, share_size, expected_share_hash, reverify_count)
+		VALUES (?, ?, ?, ?, ?, ?)
+		ON CONFLICT(node_id)
+		DO UPDATE SET reverify_count = pending_audits.reverify_count + 1`,
+	)
+	_, err := containment.db.ExecContext(ctx, statement,
+		pendingAudit.NodeID.Bytes(), pendingAudit.PieceID.Bytes(), pendingAudit.StripeIndex, pendingAudit.ShareSize, pendingAudit.ExpectedShareHash, pendingAudit.ReverifyCount,
+	)
+	if err != nil {
+		return audit.ContainError.Wrap(err)
+	}
+	return nil
+}
+
+// Delete deletes the pending audit
+func (containment *containment) Delete(ctx context.Context, id pb.NodeID) error {
+	if id.IsZero() {
+		return audit.ContainError.New("node ID empty")
+	}
+	isDeleted, err := containment.db.Delete_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return audit.ErrContainedNotFound.New(id.String())
+	}
+	if err != nil {
+		return audit.ContainError.Wrap(err)
+	}
+	if !isDeleted {
+		return audit.ErrContainDelete.New(id.String(), audit.ContainError.Wrap(err))
```

Comment:
At this moment `err` should be `nil`. Wrapping it will have no effect - the result will be still `nil`.

Also, see the comment above that I think we should not return an error in this case.

Reply:
should I just remove this case and be done?

Reply:
Yes. In case you want to give info to the caller if there was something deleted or not, you can make the method return `isDeleted` boolean and just pass it over.